flexible_event_config.md

```diff
@@ -169,6 +170,15 @@ When the `event_report_window` for a spec completes, we will map its summary val
   "trigger_summary_bucket": [<bucket start>, <bucket end>]
 }
 ```
+### Trigger prioritization
+
+Given that triggering attribution can affect a source's state without producing a report, we will need a new algorithm for doing trigger prioritization. Here is a sketch of how it could work:
+
+1. For every source, maintain a list of triggers, sorted in order of priority (descending), then trigger time (ascending)
+2. At the end of any report window (across all of a source's specs, breaking ties arbitrarily):
+    1. Iterate through the source's triggers in order, "applying" them to generate a list of "speculative" reports. Stop when privacy limits are hit.
+    2. Flush all of the speculative reports that are scheduled to be emitted in the current window, and update the source's state based on all of the triggers that were successfully applied.
```

Comment:
Step 2.2 seems to call for deleting all the reports generated in step 2.1.

Reply:
Yep I think that's good to clarify: How about this 
```suggestion
    2. Flush all of the speculative reports that are scheduled to be emitted in the current window, and update the source's state based on all of the triggers that were successfully applied. Delete all of the speculative reports that are not scheduled to be emitted in this window.
```

Reply:
Oh, maybe I misunderstood. Does "flush" mean delete or send?

Reply:
If it means "send" then it's fine.

Reply:
It means send! I don't know why I used such an ambiguous word 🤦

Reply:
```suggestion
    2. Send all of the speculative reports that are scheduled to be emitted in the current window, and update the source's state based on all of the triggers that were successfully applied. Delete all of the speculative reports that are not scheduled to be emitted in this window.
```